bootstrap/internal/controller/infraenv_controller.go

```diff
@@ -77,10 +101,30 @@ func (r *InfraEnvReconciler) attachISOToAgentBootstrapConfigs(ctx context.Contex
 		}
 
 		// Add ISO to agentBootstrapConfig status
-		agentBootstrapConfig.Status.ISODownloadURL = infraEnv.Status.ISODownloadURL
+		agentBootstrapConfig.Status.ISODownloadURL = downloadURL
 		if err := r.Client.Status().Update(ctx, &agentBootstrapConfig); err != nil {
 			return errors.Wrap(err, "failed to update agentbootstrapconfig")
 		}
 	}
 	return nil
 }
+
+func (r *InfraEnvReconciler) getISOURL(ctx context.Context, originalURL string) (string, error) {
+	if !r.Config.UseInsecureImageURL {
+		return originalURL, nil
+	}
+
+	svc := &corev1.Service{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: r.Config.ImageServiceName, Namespace: r.Config.ImageServiceNamespace}, svc); err != nil {
+		return "", errors.Wrap(err, "failed to find assisted image service service")
+	}
+
+	_, remainderURL, found := strings.Cut(originalURL, "byapikey") //TODO: will the URL always have this?
+	if !found {
+		return "", fmt.Errorf("failed to parse InfraEnv Download URL %s", originalURL)
+	}
+
+	//TODO: better way to build this string
+	downloadURL := "http://" + svc.Spec.ClusterIP + ":" + strconv.Itoa(int(svc.Spec.Ports[0].Port)) + "/byapikey" + remainderURL
```

Comment:
You should do all of this with the standard library [net/url](https://pkg.go.dev/net/url@go1.22.5)

Reply:
Thank you!!

Reply:
Modified to use the std lib net/url package!